conformance-tests/README.md

```diff
@@ -0,0 +1,54 @@
+# Running [Gateway Conformance Tests](https://gateway-api.sigs.k8s.io/concepts/conformance/#3-conformance-tests) in kind
```

Comment:
let's collapse `conformance` and `conformance-tests` into a single directory - they are related.

```
- conformance
  - tests // contains the current conformace-tests folder contents
  - provisioner // contains the current conformance/provisioner folder contents
```

Reply:
I we expect Makefile to make changes in provisioner yaml file then it should live in `/conformance` with ReadMe and `/conformance/tests` should only have test dockerfile and runner go file